File: workers/templates/javascript/two-sum.js
const readline = require('readline');

/***USER_CODE***/

function runTestCase(rawInput) {
    const lines = rawInput.trim().split('\n');
    if (lines.length < 3) return "";
    const n = +lines[0];
    const nums = lines[1].split(' ').map(Number);
    const target = +lines[2];

    const ans = twoSum(nums, target);
    return Array.isArray(ans) ? ans.join(' ') : String(ans);
}

function main() {
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout,
        terminal: false
    });

    rl.on('line', (line) => {
        const trimmed = line.trim();
        if (!trimmed) return;
        if (trimmed === 'EXIT') {
            rl.close();
            process.exit(0);
        }

        const spaceIdx = trimmed.indexOf(' ');
        if (spaceIdx === -1) return;

        const caseId = trimmed.substring(0, spaceIdx);
        const b64Input = trimmed.substring(spaceIdx + 1);
        let rawInput = '';
        try {
            rawInput = Buffer.from(b64Input, 'base64').toString('utf8');
        } catch (_) {
            rawInput = b64Input;
        }

        // Intercept user console output to protect judge protocol channel
        const originalLog = console.log;
        const originalError = console.error;
        const originalWarn = console.warn;
        const originalInfo = console.info;
        let userDebugOutput = '';

        console.log = (...args) => { userDebugOutput += args.map(String).join(' ') + '\n'; };
        console.error = (...args) => { userDebugOutput += args.map(String).join(' ') + '\n'; };
        console.warn = (...args) => { userDebugOutput += args.map(String).join(' ') + '\n'; };
        console.info = (...args) => { userDebugOutput += args.map(String).join(' ') + '\n'; };

        try {
            const output = runTestCase(rawInput);
            const b64Output = Buffer.from(String(output), 'utf8').toString('base64');
            process.stdout.write(`${caseId} OK ${b64Output}\n`);
        } catch (err) {
            const errMsg = err && err.stack ? err.stack : (err ? err.message : 'Runtime Error');
            const b64Err = Buffer.from(String(errMsg), 'utf8').toString('base64');
            process.stdout.write(`${caseId} ERROR ${b64Err}\n`);
        } finally {
            // Restore console methods
            console.log = originalLog;
            console.error = originalError;
            console.warn = originalWarn;
            console.info = originalInfo;
        }
    });
}

main();

File: workers/templates/python/two-sum.py
import sys
from io import StringIO
import base64

/***USER_CODE***/

def parse_input(raw_input):
    tokens = raw_input.strip().split()
    if len(tokens) < 2:
        return [], 0
    n = int(tokens[0])
    nums = [int(tokens[i+1]) for i in range(n)]
    target = int(tokens[n+1])
    return nums, target

def run_test_case(raw_input):
    nums, target = parse_input(raw_input)
    result = Solution().twoSum(nums, target)
    return ' '.join(str(x) for x in result) if isinstance(result, (list, tuple)) else str(result)

def main():
    for line in sys.stdin:
        trimmed = line.strip()
        if not trimmed:
            continue
        if trimmed == 'EXIT':
            sys.exit(0)

        space_idx = trimmed.find(' ')
        if space_idx == -1:
            continue

        case_id = trimmed[:space_idx]
        b64_input = trimmed[space_idx + 1:]

        try:
            raw_input = base64.b64decode(b64_input).decode('utf-8')
        except:
            raw_input = b64_input

        try:
            output = run_test_case(raw_input)
            b64_output = base64.b64encode(output.encode('utf-8')).decode('utf-8')
            sys.stdout.write(f'{case_id} OK {b64_output}\n')
            sys.stdout.flush()
        except Exception as err:
            err_msg = str(err) if str(err) else 'Runtime Error'
            b64_err = base64.b64encode(err_msg.encode('utf-8')).decode('utf-8')
            sys.stdout.write(f'{case_id} ERROR {b64_err}\n')
            sys.stdout.flush()

if __name__ == '__main__':
    main()
